app/validator/twitter.js

```diff
@@ -1,91 +1,124 @@
-'use strict';
-
 const _ = require('lodash');
 const logger = require('app/logger');
 const config = require('app/config');
-const TwitterApi = require('twitter');
+const { TwitterClient } = require('twitter-api-client');
+
 const Validator = require('app/validator/base');
 
 const { ValidatorEvent } = require('app/validator/events');
 const { RequestJudgementCollection } = require('app/db');
 
+const utils = require('app/utils');
+
 class TwitterValidator extends Validator {
     constructor(config) {
         super(config);
         this._is_twitter_verified = undefined;
-    }
-
-    async invoke(userName, walletAddr) {
-        const client = new TwitterApi({
-            consumer_key: this.config.consumerKey,
-            consumer_secret: this.config.consumerSecret,
-            access_token_key: this.config.tokenKey,
-            access_token_secret: this.config.tokenSecret,
+        this.client = new TwitterClient({
+            apiKey: this.config.apiKey,
+            apiSecret: this.config.apiKeySecret,
+            accessToken: this.config.accessToken,
+            accessTokenSecret: this.config.accessTokenSecret,
+            disableCache: true,
+            maxByteSize: 32000000,
+            ttl: 360,
         });
+    }
 
-        logger.debug('Twitter verification task starts running ...');
-        // Save context of current function
-        const self = this;
-        const caller = setInterval(async () => {
-            const resp = await self._poll(client, userName, walletAddr);
-            if (resp) {
-                logger.debug(
-                    `Twitter verification for user ${userName}, account ${walletAddr} passed, clear interval...`
-                );
-                // TODO Manipulate database, the following code may result in bugs
-                await RequestJudgementCollection.setTwitterVerifiedSuccess(walletAddr, userName);
-                clearInterval(caller);
-                clearTimeout(timeoutGuard);
-            } else {
-                logger.debug(`Retry polling twitter message for user ${userName}, account ${walletAddr}...`);
+    async invoke(twitterAccount, token) {
+        const link = `${this.config.callbackEndpoint}?token=${token}`;
+        try {
+            const resp = await this.sendCtaMessage(twitterAccount, link);
+            logger.debug(`Send verification message to ${JSON.stringify(resp)} successfully.`);
+            return resp;
+        } catch (error) {
+            const errorData = JSON.parse(error.data);
+            /* eslint-disable-next-line */
+            if (errorData.hasOwnProperty('errors')) {
+                for (let e of errorData.errors) {
+                    if (e.code === 17) {
```

Comment:
If I understand it correctly, by default we cannot send DM to any user that hasn't followed us. So this is a very common case I guess.

Reply:
Yes. We should let the user follow us first. 
Or the user allows receiving any private conversation in his privacy settings. 
The biggest problem is that the user turns off this setting by default. So it backs to the original problem, we have to let the user search our official Twitter account.